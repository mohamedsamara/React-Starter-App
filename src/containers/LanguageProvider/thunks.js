/*
 *
 * LanguageProvider thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleSetLocale } from './actions';

export function setLocale(lang) {
  return function(dispatch, getState) {
    localStorage.appLang = lang;
    return dispatch(handleSetLocale(lang));
  };
}
