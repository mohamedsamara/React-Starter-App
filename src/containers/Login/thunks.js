/*
 * 
 * Login
 * 
 */


// Do this in every file where you use `fetch`
import fetch from 'cross-fetch'
import validator from "validator";
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';


import {} from './actions';


/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
