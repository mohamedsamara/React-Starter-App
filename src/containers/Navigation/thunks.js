/*
 *
 * Navigation
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import { handleToggleDrawer } from './actions';

// Toggle Drawer
export function toggleDrawer(side, open) {
  const isDrawerOpen = {};
  isDrawerOpen.side = side;
  isDrawerOpen.open = open;

  return function(dispatch, getState) {
    return dispatch(handleToggleDrawer(isDrawerOpen));
  };
}
