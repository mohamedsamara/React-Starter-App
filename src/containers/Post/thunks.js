/*
 *
 * Post thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import {
  fetchPostsRequest,
  fetchPostsSuccess,
  handlePostFailure,
  deletePostSuccess,
} from './actions';

// post failure handler
export function postFailureHandler(errorData) {
  return function(dispatch, getState) {
    return dispatch(handlePostFailure(errorData));
  };
}

// fetch posts API
export function fetchPosts() {
  return function(dispatch, getState) {
    dispatch(fetchPostsRequest(true));

    return fetch('https://jsonplaceholder.typicode.com/posts', {
      method: 'GET',
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        toastr.success('Success', 'Posts fetched successfully');
        dispatch(fetchPostsSuccess(data));
        dispatch(fetchPostsRequest(false));
      })
      .catch(error => {
        const errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to fetch Posts';

        toastr.error('Error', 'Error fetching todos');

        dispatch(postFailureHandler(errorData));
      });
  };
}

// delete post API
export function deletePost(id, index) {
  return function(dispatch, getState) {
    return fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
      method: 'DELETE',
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        // check if the post is deleted then dispatch an action to remove the post from the store.
        toastr.success('Success', 'Post deleted successfully');
        dispatch(deletePostSuccess(index));
      })
      .catch(error => {
        const errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to delete post';

        toastr.error('Error', 'Error deleting post');

        dispatch(postFailureHandler(errorData));
      });
  };
}
