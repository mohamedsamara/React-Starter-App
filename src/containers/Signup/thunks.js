/*
 *
 * Signup thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import {} from './actions';

/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
