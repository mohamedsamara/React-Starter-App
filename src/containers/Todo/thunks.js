/*
 *
 * Todo
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import {
  fetchTodosRequest,
  fetchTodosSuccess,
  fetchTodosFailure
} from './actions';

export function fetchTodos() {
  return function(dispatch, getState) {
    dispatch(checkTodos());

    return fetch('https://jsonplaceholder.typicode.com/todos', {
      method: 'GET'
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        toastr.success('Success', `Todos Fetched Successfully`);
        dispatch(fetchTodosSuccess(data));
        dispatch(fetchTodosRequest(false));
      })
      .catch(error => {
        let errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to fetch Todos';

        toastr.error('Error', `Error Fetching Todos`);

        dispatch(fetchTodosFailure(errorData));
      });
  };
}

// Check Todos
export function checkTodos() {
  return function(dispatch, getState) {
    if (getState().todo.todos.length == 0) {
      dispatch(fetchTodosRequest(true));
    }
  };
}
