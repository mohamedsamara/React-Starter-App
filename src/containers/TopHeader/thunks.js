/*
 *
 * TopHeader thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleOpenMenu, handleCloseMenu } from './actions';

// Open Menu
export function openMenu(target) {
  return function(dispatch, getState) {
    return dispatch(handleOpenMenu(target));
  };
}

// Close Menu
export function closeMenu() {
  return function(dispatch, getState) {
    return dispatch(handleCloseMenu());
  };
}
