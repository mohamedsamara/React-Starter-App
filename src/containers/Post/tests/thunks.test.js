import { deletePost } from '../thunks';

describe('Posts Thunk', () => {
  it('should dispatch correct actions when deleting post', () => {
    const mockDispatch = jest.fn();
    const result = deletePost(1);

    expect(typeof deletePost(1)).toEqual('function');

    return result(mockDispatch).then(() => {
      expect(mockDispatch.mock.calls).toMatchSnapshot();
    });
  });
});
