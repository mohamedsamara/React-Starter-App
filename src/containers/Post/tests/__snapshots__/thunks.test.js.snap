// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Thunk should dispatch correct actions when deleting post 1`] = `
Array [
  Array [
    Object {
      "index": undefined,
      "type": "src/Post/DELETE_POST_SUCCESS",
    },
  ],
]
`;
